messaging/runtime/src/main/java/io/quarkiverse/flow/messaging/ContentBasedRouterEventsPublisher.java

```diff
@@ -0,0 +1,58 @@
+package io.quarkiverse.flow.messaging;
+
+import java.util.concurrent.CompletableFuture;
+
+import org.eclipse.microprofile.reactive.messaging.Message;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.cloudevents.CloudEvent;
+import io.cloudevents.core.provider.EventFormatProvider;
+import io.cloudevents.jackson.JsonFormat;
+import io.serverlessworkflow.impl.events.EventPublisher;
+import io.smallrye.reactive.messaging.MutinyEmitter;
+
+/**
+ * A content-based router {@link EventPublisher}.
+ *
+ * @see <a href="https://www.enterpriseintegrationpatterns.com/patterns/messaging/ContentBasedRouter.html>Messaging Patterns -
+ *      Content-Based Router</a>
+ */
+public abstract class ContentBasedRouterEventsPublisher implements EventPublisher {
+
+    protected static final String ENGINE_PREFIX = "io.serverlessworkflow";
+    private static final Logger LOG = LoggerFactory.getLogger(ContentBasedRouterEventsPublisher.class);
+    private static final JsonFormat FORMAT = (JsonFormat) EventFormatProvider.getInstance()
+            .resolveFormat(JsonFormat.CONTENT_TYPE);
+
+    @Override
+    public final CompletableFuture<Void> publish(CloudEvent event) {
+        if (!accept(event)) {
+            return CompletableFuture.completedFuture(null);
+        }
+        try {
+            byte[] structured = FORMAT.serialize(event);
+            LOG.debug("Flow: Publishing on channel {} event={}", channelName(), new String(structured));
```

Comment:
Since building the string from the byte[] might be a costly operation, I think you should wrap the LOG statement within a LOG.isDebugEnabled()